lib/extensions/conditional/active_record/base.rb

```diff
@@ -61,5 +61,8 @@ module ConditionClassMethods
     def dependent_class
       raise NotImplementedError
     end
+
+    def resolve_conditional_for(_course_user)
```

Comment:
Subclasses do not need to implement this method.

This method will contain the common code to call the dependency graph for resolving the conditionals. (mainly passing course_user to the course.dependency_graph.resolve)